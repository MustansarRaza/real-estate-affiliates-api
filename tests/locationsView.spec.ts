import { loadSequelizeFixtures, useRequestFactory } from "./support";
import { ModelFixtureNames } from "./support/data";

describe("GET /locations", () => {
  beforeEach(async () => {
    await loadSequelizeFixtures(ModelFixtureNames.COUNTRIES, ModelFixtureNames.LOCATIONS);
  });

  it("returns a flat list of locations", async () => {
    const { body, header } = await useRequestFactory()
      .get("/locations")
      .expect(200);

    expect(body).toMatchSnapshot();
    expect(header).toMatchSnapshot();
  });
});
