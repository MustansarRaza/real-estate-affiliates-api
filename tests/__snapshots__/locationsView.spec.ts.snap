// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GET /locations returns a flat list of locations 1`] = `
Array [
  Object {
    "id": 1,
    "iso31662Code": "PK",
    "name": "Pakistan",
    "parentID": null,
    "typeID": 0,
  },
  Object {
    "id": 2,
    "name": "Punjab",
    "parentID": 1,
    "typeID": 1,
  },
  Object {
    "id": 3,
    "name": "Lahore",
    "parentID": 2,
    "typeID": 2,
  },
  Object {
    "id": 4,
    "name": "Multan",
    "parentID": 2,
    "typeID": 2,
  },
  Object {
    "id": 50,
    "name": "DHA Phase-I",
    "parentID": 3,
  },
]
`;

exports[`GET /locations returns a flat list of locations 2`] = `
Object {
  "cache-control": "private, no-store",
  "connection": "close",
  "content-length": "266",
  "content-type": "application/json; charset=utf-8",
  "etag": "W/\\"10a-1CLIWjOadkiuyBDBQ+cAX3C1F/k\\"",
  "strict-transport-security": "max-age=15552000; includeSubDomains",
  "vary": "Accept-Encoding",
  "x-content-type-options": "nosniff",
  "x-dns-prefetch-control": "off",
  "x-download-options": "noopen",
  "x-frame-options": "SAMEORIGIN",
  "x-xss-protection": "1; mode=block",
}
`;
